### PR TITLE
CI: Remove dependency on packagesbuild and use native Xcode tools

### PR DESCRIPTION
**Pull request by PatTheMav**
### Description
Removes dependency on `Packages.app` and uses native `pkgbuild` and `productbuild` tools to generate installer packages for macOS.

### Motivation and Context
As reported in https://github.com/obsproject/obs-plugintemplate/issues/91, the download for `Packages.app` is unversioned and the page itself not available via HTTPs. Its developer did a patch/minor version upgrade without putting up a changelog or changing the version number (though admittedly the fix they integrated was valid).

As we only use the command line tool to generate a final installer package, using a `distribution` file generated by CMake suffices to generate those packages.

Fixes https://github.com/obsproject/obs-plugintemplate/issues/91.

### How Has This Been Tested?
Tested locally with codesigning enabled.

### Types of changes
- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
cc @gxalpha

File: cmake/macos/resources/create-package.cmake.in [hunk 1]
```diff
@@ -1,7 +1,35 @@
...
-set(UUID_INSTALLER ${UUID_INSTALLER})
-configure_file(cmake/macos/resources/installer-macos.pkgproj.in
-  "${CMAKE_CURRENT_BINARY_DIR}/installer-macos.generated.pkgproj")
+make_directory("$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/package/Library/Application Support/obs-studio/plugins")
+
+if(EXISTS "$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/@CMAKE_PROJECT_NAME@.plugin" AND NOT IS_SYMLINK "$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/@CMAKE_PROJECT_NAME@.plugin")
+  file(INSTALL DESTINATION "$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/package/Library/Application Support/obs-studio/plugins"
+    TYPE DIRECTORY FILES "$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/@CMAKE_PROJECT_NAME@.plugin" USE_SOURCE_PERMISSIONS)
+
+  if(CMAKE_INSTALL_CONFIG_NAME MATCHES "^([Rr][Ee][Ll][Ee][Aa][Ss][Ee])$" OR CMAKE_INSTALL_CONFIG_NAME MATCHES "^([Mm][Ii][Nn][Ss][Ii][Zz][Ee][Rr][Ee][Ll])$")
```
Comment (RytoEX) [inline cmake/macos/resources/create-package.cmake.in:7]: Would it be more readable to use a temp var to deal with case insensitive comparisons?
```cmake
string(TOLOWER "${CMAKE_INSTALL_CONFIG_NAME}" LOWER_CMAKE_INSTALL_CONFIG)
if(CMAKE_INSTALL_CONFIG_NAME MATCHES "^Release$" OR CMAKE_INSTALL_CONFIG_NAME MATCHES "^MinSizeRel$")
unset(LOWER_CMAKE_INSTALL_CONFIG)
```
or:
```cmake
string(TOLOWER "${CMAKE_INSTALL_CONFIG_NAME}" LOWER_CMAKE_INSTALL_CONFIG)
if(CMAKE_INSTALL_CONFIG_NAME STREQUAL "Release" OR CMAKE_INSTALL_CONFIG_NAME STREQUAL "MinSizeRel")
unset(LOWER_CMAKE_INSTALL_CONFIG)
```

No strong feelings either way.

File: cmake/macos/resources/distribution.in [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                                    'InstallationCheckError',
+                                    system.localizedString('DISTRIBUTION_TITLE')
+                                  );
+                my.result.message = ' ';
+                my.result.type = 'Fatal';
+                return false;
+            }
+        }
+    </script>
+
```
Comment (RytoEX) [inline cmake/macos/resources/distribution.in:33]: Is it common to have a blank line here?

**Comment by RytoEX** [comment]
cc @gxalpha for review

**Review by RytoEX — COMMENTED** [review]
Should this also remove the packages section from buildspec.json?
https://github.com/obsproject/obs-plugintemplate/blob/b4a4eb51a8210eaaaa1fdc6ed88c22b89e5563cd/buildspec.json#L34-L41

File: cmake/macos/resources/create-package.cmake.in (hunk 1 above)
Comment (PatTheMav) [reply cmake/macos/resources/create-package.cmake.in:7]: This follows the default code created by CMake for installation scripts (check the `cmake_install.cmake` script generated in every target directory).

File: cmake/macos/resources/distribution.in (hunk 2 above)
Comment (PatTheMav) [reply cmake/macos/resources/distribution.in:33]: That's the default "newline at end of file".

**Comment by PatTheMav** [comment]
> Should this also remove the packages section from buildspec.json?
> 
> https://github.com/obsproject/obs-plugintemplate/blob/b4a4eb51a8210eaaaa1fdc6ed88c22b89e5563cd/buildspec.json#L34-L41

It's effectively ignored by the changes, but doesn't hurt to remove it as well.

File: cmake/macos/resources/create-package.cmake.in (hunk 1 above)
Comment (RytoEX) [reply cmake/macos/resources/create-package.cmake.in:7]: Unfortunate. Okay then.

**Comment by RytoEX** [comment]
> > Should this also remove the packages section from buildspec.json?
> > https://github.com/obsproject/obs-plugintemplate/blob/b4a4eb51a8210eaaaa1fdc6ed88c22b89e5563cd/buildspec.json#L34-L41
> 
> It's effectively ignored by the changes, but doesn't hurt to remove it as well.
> 
> The UUIDs for macOS can also be removed now because they were only required by Packages, the built-in tools do not require this and use `pkg-ref` as identifier it seems.

I just don't want someone being confused by it or reporting an outdated package/tool when we don't use it.

File: cmake/macos/resources/distribution.in (hunk 2 above)
Comment (PatTheMav) [reply cmake/macos/resources/distribution.in:33]: Ah my bad, I mistook the line. That one can be safely removed.

**Comment by PatTheMav** [comment]
@RytoEX please revisit this so it can be merged soon-ish, I'd like to unbreak the repo.

**Review by RytoEX — APPROVED** [review]
Seems fine.

**Pull request merged**